src/lib/php/Dao/ClearingDao.php

```diff
@@ -756,41 +756,73 @@ public function markDirectoryAsIrrelevant(ItemTreeBounds $itemTreeBounds,$groupI
+      $sql = $uploadTreeProxy->asCte()
+          .' INSERT INTO clearing_decision (uploadtree_fk,pfile_fk,user_fk,group_fk,decision_type,scope) 
+            SELECT uploadtree_pk itemid,pfile_fk pfile_id, $'.($a+1).', $'.$a.', $'.($a+2).', $'.($a+3).' FROM UploadTreeView
+	    WHERE NOT EXISTS (SELECT uploadtree_fk FROM clearing_decision WHERE decision_type=$5 and uploadtree_fk=UploadTreeView.uploadtree_pk)';
+    }else{
+      $params[] = DecisionTypes::IRRELEVANT;
+      $sql = $uploadTreeProxy->asCte()
+          .' DELETE FROM clearing_decision WHERE clearing_decision_pk IN (SELECT clearing_decision_pk FROM clearing_decision WHERE 
+           decision_type = $'.($a+1).' AND uploadtree_fk IN (SELECT uploadtree_pk FROM UploadTreeView) ORDER BY clearing_decision.date_added DESC)';
+    }
     $this->dbManager->prepare($statementName, $sql);
     $res = $this->dbManager->execute($statementName,$params);
     $this->dbManager->freeResult($res);
-  }  
-    
+  }
+
   /**
    * @param ItemTreeBounds $itemTreeBounds
    * @param int $groupId
    * @param int $userId
    */
-  protected function markDirectoryAsIrrelevantIfUserEdited(ItemTreeBounds $itemTreeBounds,$groupId,$userId) 
+  protected function markDirectoryAsIrrelevantIfUserEdited(ItemTreeBounds $itemTreeBounds,$groupId,$userId,$removeDecision=false) 
   {
     $statementName = __METHOD__ ;
     $params = array($itemTreeBounds->getLeft(), $itemTreeBounds->getRight());
     $condition = "ut.lft BETWEEN $1 AND $2";
     $decisionsCte = $this->getRelevantDecisionsCte($itemTreeBounds, $groupId, $onlyCurrent=true, $statementName, $params, $condition);
-    $params[] = $userId; 
-    $a = count($params);
-    $params[] = $groupId;
-    $params[] = DecisionTypes::IRRELEVANT;
-    $params[] = DecisionScopes::ITEM;
-    $this->dbManager->prepare($statementName, $decisionsCte
-        .' INSERT INTO clearing_decision (uploadtree_fk,pfile_fk,user_fk,group_fk,decision_type,scope) 
-          SELECT itemid,pfile_id, $'.$a.', $'.($a+1).', $'.($a+2).', $'.($a+3).' FROM allDecs ad WHERE type_id!=$'.($a+2));
+    if(!$removeDecision){
+      $params[] = $userId; 
+      $a = count($params);
+      $params[] = $groupId;
+      $params[] = DecisionTypes::IRRELEVANT;
+      $params[] = DecisionScopes::ITEM;
+      $this->dbManager->prepare($statementName, $decisionsCte
+          .' INSERT INTO clearing_decision (uploadtree_fk,pfile_fk,user_fk,group_fk,decision_type,scope) 
+            SELECT itemid,pfile_id, $'.$a.', $'.($a+1).', $'.($a+2).', $'.($a+3).' FROM allDecs ad WHERE type_id IS NULL');
+    }else{
+      $params[] = DecisionTypes::IRRELEVANT;
+      $this->dbManager->prepare($statementName, $decisionsCte
+          .' DELETE FROM clearing_decision WHERE clearing_decision_pk IN (SELECT clearing_decision_pk FROM clearing_decision WHERE 
+           decision_type = $'.count($params).' AND uploadtree_fk IN (SELECT itemid FROM allDecs) ORDER BY clearing_decision.date_added DESC)');
```

Comment:
Maybe you could define a variable `$a = count($params)`. Then this query would look more consistent with the other queries...